esm_runscripts/filelists.py

```diff
@@ -258,13 +264,40 @@ def target_subfolders(config):
                                 descr
                             ] = filename.replace("*", source_filename)
                         elif "/" in filename:
-                            config[model][filetype + "_targets"][
-                                descr
-                            ] = "/".join(filename.split("/")[:-1]) + "/" + source_filename.split("/")[-1]
+                            config[model][filetype + "_targets"][descr] = (
+                                "/".join(filename.split("/")[:-1])
+                                + "/"
+                                + source_filename.split("/")[-1]
+                            )
                         else:
```

Comment:
Overall fix looks very nice 👏 👏 
However, I see one issue here:
This is a fix for the `<exp_path>/restart/oifs/DATE/srf0006.*` cases, where the file name ends with the glob wildcard. `else` branch would catch this but what I would do is something like this
```python
elif filename.endswith('.*'):    # NOTE-1
    # ...
    # rest of the hotfix
    # ...
else:
    # ...
    # these would get executed if nothing else is matched (like a case default situation). 
    # Eg, it could trap some typos with the wildcards
    # Maybe a warning message or a placeholder for a further bugfix
```
NOTE-1: to be on the safe side a regex pattern of `re.search("\.\*[ \t]*$", filename)` could have been used but it could also be an overkill

Reply:
I think what you are asking for is already solved here:
https://github.com/esm-tools/esm_runscripts/blob/hotfix/wildcards_renaming_to_target/esm_runscripts/filelists.py#L275-L298

The idea is that the wild cards are split in pieces using the "*" so for example `asdasd*mmm` would be split in `["asdasd", "mmm"]` and then substitution of this pieces occur on the target.